cdragontoolbox/hashes.py

```diff
@@ -328,6 +328,39 @@ def substitute_basename_words(self):
     def add_basename_word(self):
         super()._add_basename_word(self.build_wordlist())
 
+    def multi_substitution(self, plugin, fileext=""):
```

Comment:
Instead of filtering on a plugin name you should filter on a custom filter, or accept a custom list of paths as parameter. Otherwise, the method is LCU-specific.

Reply:
The method is indeed currently lcu specific, that's why it's only in the lcu hash guesser.

Reply:
If you make it generic, you could move it to the be class.